Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.cpp

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+
+#include "MultiplayerConnectionViewportMessageSystemComponent.h"
+#include <Atom/RPI.Public/ViewportContextBus.h>
+#include <Atom/RPI.Public/ViewportContext.h>
+#include <AzCore/Console/IConsole.h>
+#include <AzFramework/Entity/EntityDebugDisplayBus.h>
+#include <AzNetworking/Framework/INetworking.h>
+#include <Multiplayer/IMultiplayerSpawner.h>
+#include <Multiplayer/MultiplayerConstants.h>
+
+namespace Multiplayer
+{
+    constexpr float defaultConnectionMessageFontSize = 0.7f;
+    const AZ::Vector2 viewportConnectionBottomRightBorderPadding(-40.0f, -40.0f);
+
+    AZ_CVAR_SCOPED(bool, cl_viewportConnectionStatus, true, nullptr, AZ::ConsoleFunctorFlags::DontReplicate,
+        "This will enable displaying connection status in the client's viewport while running multiplayer.");
+
+    AZ_CVAR_SCOPED(float, cl_viewportConnectionMessageFontSize, defaultConnectionMessageFontSize, nullptr, AZ::ConsoleFunctorFlags::DontReplicate, 
+        "The font size used for displaying updates on screen while the multiplayer editor is connecting to the server.");
+    
+    void MultiplayerConnectionViewportMessageSystemComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<MultiplayerConnectionViewportMessageSystemComponent, AZ::Component>()
+                ->Version(1);
+        }
+    }
+    
+    void MultiplayerConnectionViewportMessageSystemComponent::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC_CE("MultiplayerService"));
+    }
+
+    void MultiplayerConnectionViewportMessageSystemComponent::Activate()
+    {
+        AZ::RPI::ViewportContextNotificationBus::Handler::BusConnect(
+            AZ::RPI::ViewportContextRequests::Get()->GetDefaultViewportContextName());
+    }
+
+    void MultiplayerConnectionViewportMessageSystemComponent::Deactivate()
+    {
+        AZ::RPI::ViewportContextNotificationBus::Handler::BusDisconnect();
+    }
+
+    void MultiplayerConnectionViewportMessageSystemComponent::OnRenderTick()
+    {
+        if (!cl_viewportConnectionStatus)
+        {
+            return;
+        }
+
+        AZ::RPI::ViewportContextPtr viewport = AZ::RPI::ViewportContextRequests::Get()->GetDefaultViewportContext();
+        if (!viewport)
+        {
+            return;
```

Comment:
Nit: Do we want to throw an error or warning in this case? Same for all the other checking below.

Reply:
Fixed! Making them WarnOnce since this is happening ontick; it might be valid to simply stop rendering entirely if these checks return null, but some of these are using old cry systems so i'm not sure about their life cycle so I'm okay with just keepign a warning.

Reply:
Good call out! Hmm, this is also running on ServerLauncher which wont have a renderer so logging warnings causes spam (even AZWarnOnce is getting called enough times to become noticeable, not sure why)
Also, the font renderers life cycle isn't guaranteed (looks like its using cry systems); until that changes, it might be best to leave the checks inside OnTick, but not warn in order to avoid spam :c

Reply:
I found a way to clean this up so the system is better at coming alive when needed instead of waiting around. It’ll require some more callbacks for MP system, I’ll add that to a new Pr.